src/tools/database/sql.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Tool,
+  ToolInput,
+  ToolError,
+  BaseToolOptions,
+  BaseToolRunOptions,
+  JSONToolOutput,
+} from "@/tools/base.js";
+import { z } from "zod";
+import { Sequelize, Options } from "sequelize";
+import { getMetadata } from "@/tools/database/metadata.js";
+import { Cache } from "@/cache/decoratorCache.js";
+
+type Provider = "mysql" | "mariadb" | "postgres" | "mssql" | "db2" | "sqlite" | "oracle";
+
+interface ToolOptions extends BaseToolOptions {
+  provider: Provider;
+  connection: Options;
+}
+type ToolRunOptions = BaseToolRunOptions;
+
+export class SQLTool extends Tool<JSONToolOutput<any>, ToolOptions, ToolRunOptions> {
+  name = "SQLTool";
+
+  description =
+    "Converts natural language to SQL query and executes it using the provided tables schema.";
+
+  inputSchema() {
+    return z.object({
+      query: z.string({ description: "The SQL query to be executed." }).min(1),
+    });
+  }
+
+  static {
+    this.register();
+  }
+
```

Comment:
Add a constructor and verify if the given schema is `supported.` It is better to throw an error as soon as possible.